### ISSUE TITLE
no result when running v0.8.1 (with worker or from cdn)

### ISSUE THREAD

**Issue by bassjobsen** [medialize/sass.js#29]
First i tried v0.6.3 from CDN:

```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Sass in browser</title>
    <script src="//code.jquery.com/jquery-2.1.4.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/sass.js/0.6.3/sass.min.js"></script>
    <!--script src="bower_components/sass.js/dist/sass.js"></script-->
    <style id="csscode"></style>
    <script>

      //Sass.initialize('bower_components/sass.js/dist/sass.worker.js');
      function compileCSS(color) {
      //Sass.compile('$color: ' + color +  '; h1 { color: $color; }', function(result){
      //console.log('result: ' + result);    
      //});
      var css = Sass.compile('$color: ' + color +  '; h1 { color: $color; }');
      console.log(css);
      }
      compileCSS('orange');
      console.log('done......');
    </script>

</head>
<body>
</body>
</html>
```

The above writes to the console as expected:

>SyntaxError: "use asm" is only meaningful in the Directive Prologue of a function body sass.min.js:1982:107
>"pre-main prep time: 39 ms" sass.min.js:55:137
>"h1 {
>  color: orange; }
>" t:20:6
>"done......"

Then i replace `//cdnjs.cloudflare.com/ajax/libs/sass.js/0.6.3/sass.min.js` with `//cdnjs.cloudflare.com/ajax/libs/sass.js/0.6.3/sass.min.js` now the output is as follows:

> TypeError: b._worker is null

Finally i also tried: `bower install sass.js` -> `bower sass.js#~0.8.1           install sass.js#0.8.1` and after that i use in the HTML:

    <script src="bower_components/sass.js/dist/sass.js"></script>

and
 
     Sass.initialize('bower_components/sass.js/dist/sass.worker.js');

the latest setting only sets `css` to undefined:

>undefined t:19:6
>"done......" t:22:6
>SyntaxError: "use asm" is only meaningful in the Directive Prologue of a function body >sass.worker.js:1907:60

So how can i use my code with version 0.8.1 in browser? Notice using `Sass.compile(source, callback)` <strike>also result in `undefined`</strike>;

      Sass.compile('$color: ' + color +  '; h1 { color: $color; }',function(result) {      
      console.log('result:' + result);
      });

Does not generate any output other than:

>"done......" t:25:6
>SyntaxError: "use asm" is only meaningful in the Directive Prologue of a function body

**Comment by rodneyrehm** [comment]
Maybe looking at [sass.worker.html](http://medialize.github.io/sass.js/sass.worker.html) helps? 

```js
Sass.compile('$color: #123; h1 { color: $color; }', function(result) {      
  // result is an object, don't cast it to string!
  console.log('result', result);
});
```

yields the following `result`:

```js
{
  "status": 0,
  "text": "h1 {\n  color: #123; }\n",
  "map": {
    "version": 3,
    "sourceRoot": "root",
    "file": "stdout",
    "sources": [
      "stdin"
    ],
    "sourcesContent": [
      "$color: #123; h1 { color: $color; }"
    ],
    "mappings": "AAAc,EAAE,CAAC;EAAE,KAAK,EAAhB,IAAI,GAAM",
    "names": []
  },
  "files": []
}
```

**Comment by bassjobsen** [comment]
After compiling from the latest source (see issue #30):
   
    <script src="sass.js/dist/sass.js"></script>
    <script>
        Sass.initialize('sass.js/dist/sass.worker.js');
        
        Sass.writeFile('testfile.scss', '@import "sub/deeptest";\n.testfile { content: "loaded"; }', function() {
          console.log('wrote "testfile.scss"');
        });
        Sass.writeFile('sub/deeptest.scss', '.deeptest { content: "loaded"; }', function() {
          console.log('wrote "sub/deeptest.scss"');
        });
        Sass.options({ style: Sass.style.expanded }, function(result) {
          console.log("set options");
        });
        Sass.compile('@import "testfile";', function(result) {
          console.log("compiled", result.text);
        });
    </script>
    
Outputs nothing else than:    

> SyntaxError: "use asm" is only meaningful in the Directive Prologue of a function body

I also tried `sass.sync.js`:

    <script src="sass.js/dist/sass.sync.js"></script>
    <script>
      var scss = '$someVar: 123px; .some-selector { width: $someVar; }';
      Sass.compile(scss, function(result) {
        console.log(result);
      });
    </script>
    
Which outputs notging at all(, but seems to take many resources??)

**Comment by rodneyrehm** [comment]
At the time you called `.compile()` emscripten was not ready yet. I've just patched in support to delay .compile() calls until the runtime is fully initialized. Please pull the changes and recompile your local copy.

**Comment by bassjobsen** [comment]
Thanks, good job. This works!!!

**Comment by rodneyrehm** [comment]
sweet. were you doing anything special that you compiled sass.js yourself, or are you waiting for 0.8.2 to be released?

**Comment by bassjobsen** [comment]
Well, i was creating a demo inspired on http://stackoverflow.com/a/28429153/1596547 for my new book about Sass (and Compass). I found running sass.js in browser didn't worked as expected (i test before only with version 0.6.3) Then i thought that possible something was wrong with the CDN and/ or dist/ versions so i decided to compile from source to make sure.

Now my demo works:

    <!doctype html>
    <html lang="en">
      <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>Sass in browser</title>
        <script src="//code.jquery.com/jquery-2.1.4.min.js"></script>
        <script src="sass.js/dist/sass.js"></script>
        <style id="csscode"></style>
        <script>
          Sass.initialize('sass.js/dist/sass.worker.js');
          function compileCSS(color) {
          var options = {};
          Sass.compile('$color: ' + color +  '; h1 { color: $color; }', function(result){
              $('#csscode').html(result.text);
          }); 
          }
          compileCSS('orange');      
        </script>
    </head>
    <body>
    <h1>Colored text</h1>
    <form>
      Enter color name and hit enter:
      <input type="text" value="orange" name="color" />
      <input type="submit" />
    </form>
    <script>
          $( 'form' ).submit(function(event) {
          color = $( 'input[name="color"]' ).val();
          compileCSS(color); 
          event.preventDefault();
          });   
    </script>    
    </body>
    </html>

Hopefully i can replace the local versions of sass.js and sass.worker.js with those from CDN later on?

Notice that sass.sync.js  does not work in browser still. (possible i should not expect it does??). I already found that i have to move (or link) `libsass.js.mem` to a location relative to my document root.

Problem now due to:
>Successfully compiled asm.js code (total compilation time 7684ms; unable to cache asm.js in synchronous scripts; try loading asm.js via <script async> or createElement('script'))

**Comment by rodneyrehm** [comment]
> Hopefully i can replace the local versions of sass.js and sass.worker.js with those from CDN later on?

yes. Since we haven't changed the API yet, it's likely going to be released as 0.8.2

> Notice that sass.sync.js does not work in browser still. (possible i should not expect it does??). I already found that i have to move (or link) libsass.js.mem to a location relative to my document root.

I just fixed that as well. Should be working fine in every scenario now.

That said, you still need to make sure the `/dist/libsass.js.mem` is available. The sync variant is intended for use in environments like Node. In the browser you should always use the worker.

> Successfully compiled asm.js code (total compilation time 7684ms; unable to cache asm.js in synchronous scripts; try loading asm.js via 

what's the rest of that message and where's that coming from?

**Comment by bassjobsen** [comment]
Excuses me, that was the wrong (old)error. I found now:

> TypeError: asm.js type error: Disabled by debugger 

also found here: http://stackoverflow.com/questions/23453712/typeerror-asm-js-type-error-disabled-by-debugger
> "pre-main prep time: 416 ms"

both from sass.sync.js

**Comment by rodneyrehm** [comment]
Since that warning is not popping up for sass.sync.js, I figure it's closure-compiler that's screwing things up a bit. upgrading to the newest version did not solve the problem either. UglifyJS2 produces a larger file and screws things up even more. Next step is to look at minifying the sass.js code separately from the emscripten generated code (which already is minified).

**Comment by rodneyrehm** [comment]
fixed with [v0.8.2](https://github.com/medialize/sass.js/releases/tag/v0.8.2)
